modules/login/php/requestaccount.class.inc

```diff
@@ -0,0 +1,200 @@
+<?php
+/**
+ * Implements the LORIS request account page.
+ *
+ * PHP version 5
+ *
+ * @category   Behavioural
+ * @package    Main
+ * @subpackage Imaging
+ * @author     Dave MacFarlane <david.macfarlane2@mcgill.ca>
+ * @license    http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link       https://www.github.com/aces/Loris/
+*/
+namespace LORIS\login;
+
+/**
+ * Implements the LORIS request account page.
+ *
+ * @category   Behavioural
+ * @package    Main
+ * @subpackage Imaging
+ * @author     Dave MacFarlane <david.macfarlane2@mcgill.ca>
+ * @license    http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link       https://www.github.com/aces/Loris/
+ */
+class RequestAccount extends \NDB_Form
+{
+    /**
+     * Determine whether the user has permission to view this page
+     *
+     * @return bool whether the user has access
+     */
+    function _hasAccess()
+    {
+        return true;
+    }
+
+    /**
+     * Sets up the form elements on the page.
+     *
+     * @return none
+     */
+    function setup()
+    {
+        $this->addBasicText(
+            "firstname",
+            "First name",
+            array("placeholder" => "First Name")
+        );
+        $this->addBasicText(
+            "lastname",
+            "Last name",
+            array("placeholder" => "Last Name")
+        );
+        $this->addBasicText(
+            "from",
+            "Email address",
+            array("placeholder" => "Email address")
+        );
+        $this->addSelect(
+            "site",
+            "Site",
+            array_merge(
+                ["" => "Choose your site:"],
+                \Utility::getSiteList()
+            )
+        );
+        $this->addCheckbox("examiner", "Examiner role", array());
+        $this->addCheckbox("radiologist", "Radiologist", array());
+    }
+
+    /**
+     * _validate validates the data that was submitted by the user
+     *
+     * @param array $values The values that were submitted to the page.
+     *
+     * @return a (possibly empty) array of errors
+     */
+    function _validate($values)
+    {
+        $errors = array();
+        // FIXME: These error checks are direct ports from when the request
+        // account page was a standalone php script (with the ridiculous
+        // "name must be at least 3 characters" check removed, because some
+        // people don't have three character long names..) The "required" checks
+        // should probably be replaced with a required rule.
+        if (empty($values['firstname']) || $values['firstname'] == 'Bob') {
```

Comment:
What is wrong about been named Bob?

Reply:
At least there should be a message saying: We know your real name is Robert, Robert.

Reply:
I don't know. Why is Bob always so mean to Megabyte?